dds_web/web/user.py

```diff
@@ -267,8 +275,10 @@ def confirm_2fa():
 
     next_target = flask.request.args.get("next")
     # is_safe_url should check if the url is safe for redirects.
-    if next_target and not dds_web.utils.is_safe_url(next_target):
-        return flask.abort(400)
+    if next_target:
+        res = [ep for ep in VALID_ENDPOINTS if (ep in next_target)]
+        if res and not dds_web.utils.is_safe_url(next_target):
+            return flask.abort(400)
```

Comment:
This is just me thinking

```suggestion
    if next_target and (next_target in VALID_ENDPOINTS or next_target.startswith("/activate_totp")) and dds_web.utils.is_safe_url(next_target):
           # put all the stuff from line 284 to end of endpoint in here? 
    else:
            return flask.abort(400)
```
or reverse it? I dunno.